File: src/dataset/javascript/sieve_exercism-ThomasZumsteg/claude/ClaudeSonnet4_sieve_v4.js
var Sieve = function(size) {
	if (size < 2) {
		this.primes = [];
		return;
	}
	
	this.primes = [2];
	var sieve = new Array(size + 1).fill(true);
	sieve[0] = sieve[1] = false;
	
	var limit = Math.sqrt(size);
	
	for (var n = 3; n <= limit; n += 2) {
		if (sieve[n]) {
			for (var m = n * n; m <= size; m += n) {
				sieve[m] = false;
			}
		}
	}
	
	for (var i = 3; i <= size; i += 2) {
		if (sieve[i]) {
			this.primes.push(i);
		}
	}
}

module.exports = Sieve;

File: src/dataset/Java/sieve_exercism-ThomasZumsteg/claude/ClaudeSonnet4_Sieve_v4.java
import java.util.List;
import java.util.ArrayList;

public class Sieve {
    private boolean[] sieve;

    public Sieve(int limit) {
        sieve = new boolean[limit + 1];
        if (limit >= 2) {
            sieve[0] = sieve[1] = true;
            for (int i = 4; i < sieve.length; i += 2) {
                sieve[i] = true;
            }
            
            int sqrtLimit = (int) Math.sqrt(limit);
            for (int i = 3; i <= sqrtLimit; i += 2) {
                if (!sieve[i]) {
                    for (int m = i * i; m < sieve.length; m += 2 * i) {
                        sieve[m] = true;
                    }
                }
            }
        }
    }

    public List<Integer> getPrimes() {
        List<Integer> primes = new ArrayList<>();
        if (sieve.length > 2) {
            primes.add(2);
            for (int i = 3; i < sieve.length; i += 2) {
                if (!sieve[i]) {
                    primes.add(i);
                }
            }
        }
        return primes;
    }
}